### PR TITLE
Bugfix/escaped character parsing

### PR DESCRIPTION
**Pull request by chibidev**
I am building a tool that I tried to use on a larger C++ codebase, and parts of it depends on rapidjson. Apparently there's a macro definition that confuses the preprocessor (that parses macros, includes and comments) and leaves it in an unwanted state (it parses the following couple lines as part of a character literal).

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -31,10 +31,14 @@ set(tests
         integration.cpp
         libclang_parser.cpp
         parser.cpp
+        preprocessor.cpp
         visitor.cpp)
 
+    message(${CMAKE_CURRENT_LIST_DIR})
```
Comment (foonathan) [inline test/CMakeLists.txt]: Could you get rid of that message?

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -31,10 +31,14 @@ set(tests
...
+
 add_executable(cppast_test test.cpp test_parser.hpp ${tests})
 target_include_directories(cppast_test PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
+target_include_directories(cppast_test PRIVATE ${CMAKE_CURRENT_LIST_DIR}/../src)
```
Comment (foonathan) [inline test/CMakeLists.txt]: Why is this necessary?

File: test/CMakeLists.txt (hunk 1 above)
Comment (chibidev) [reply test/CMakeLists.txt]: Ah sorry, I forgot to remove.

File: test/CMakeLists.txt (hunk 2 above)
Comment (chibidev) [reply test/CMakeLists.txt]: It tests the cppast::detail::preprocess function, which is not exported publicly, just an inner library function.

Comment (foonathan) [reply test/CMakeLists.txt]: Ah, right.

File: test/preprocessor.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#include <catch.hpp>
+#include <fstream>
+
+#include "libclang/preprocessor.hpp"
+#include "test_parser.hpp"
+
+using namespace cppast;
+
+TEST_CASE("preprocessor_parses_escaped_character")
```
Comment (foonathan) [inline test/preprocessor.cpp]: Tests are failing on old libclang where I can't handle includes. Could you add `"[!hide][clang4]"` tags to the test case?

Comment (chibidev) [reply test/preprocessor.cpp]: Done :)
